core/ordersync/ordersync.go

```diff
@@ -458,6 +462,31 @@ func parseResponseWithSubprotocol(subprotocol Subprotocol, providerID peer.ID, r
 	}, nil
 }
 
+type requestMetadataForAllSubprotocols struct {
+	OrderFilter  *orderfilter.Filter `json:"orderfilter"`
+	Page         int                 `json:"page"`
+	SnapshotID   string              `json:"snapshotID"`
+	MinOrderHash common.Hash         `json:"minOrderHash"`
+}
+
+func (s *Service) createFirstRequestForAllSubprotocols() (*rawRequest, error) {
```

Comment:
I don't think it is a good idea to combine data for each subprotocol into a single struct/object. It makes it harder to introduce new subprotocols in the future in a forwards-compatible way (e.g. because two different subprotocols can no longer use the same field name in their metadata without having the same semantics). Can we separate this by subprotocol and ideally have each subprotocol implementation be responsible for generating its own first request?

Reply:
Good point. I was able to side-step this issue by leveraging some of the properties of the json.Unmarshal library relating to arrays.